### PR TITLE
fix building for MSVC

### PR DESCRIPTION
**Pull request by Jan200101**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -249,6 +249,7 @@ set_target_properties(kqueue PROPERTIES SOVERSION 0)
 if(WIN32)
   target_compile_definitions(kqueue PRIVATE _USRDLL;_WINDLL)
   target_compile_definitions(kqueue PRIVATE _CRT_SECURE_NO_WARNINGS)
+  target_compile_definitions(kqueue PRIVATE WIN32_LEAN_AND_MEAN)
```
Comment (Jan200101) [inline CMakeLists.txt:252]: the `windows.h` include in debug brought in the whole kitchen sink so  we define this globally as to not redefine any structs

File: src/common/debug.h [hunk 2]
```diff
@@ -22,6 +22,7 @@
 #include <stdbool.h>
 #ifdef _WIN32
 # include <windows.h>
+# include "../windows/platform.h"
```
Comment (Jan200101) [inline src/common/debug.h:25]: `windows/platform.h` defines the `pthread_mutex_t` that we used in this file

**Review by Jan200101 — COMMENTED** [review]
This was compiled against Visual Studio 17 2022 using
MSVC 19.31.31104.0
Windows SDK version 10.0.19041.0
Windows 10.0.19044

**Comment by arr2036** [comment]
These look fine.  As a next step you should add a matrix/entry to the Windows CI job, that way you'll guarantee the build stays functional under MSVC. If you wanted to add build instructions those would be appreciated too :)

**Comment by Jan200101** [comment]
will work on that

**Pull request merged**